reverse_strand.py

```diff
@@ -26,14 +39,72 @@
     name = read.query_name
     info = name.split("|")
     nucl = info[3]
+    if read.is_unmapped:  # Can be decoded with bitwise flag with & 4 (4 means unmapped)
+        unmappedCount += 1
+        totalCount += 1
+        continue  # `not(read.is_unmapped)` doesn't work, the unmapped reads still get through, so a continue is needed
+    if read.is_secondary:  # We only want to deal with primary reads
+        continue
+    totalCount += 1
     # Mapped onto reverse strand:
-    if read.is_reverse: # Can be decoded with bitwise flag with & 16
+    if read.is_reverse:  # Can be decoded with bitwise flag with & 16
         nucl = Seq(nucl, generic_dna).complement()
-    # Write it all to the file:
-    if read.is_unmapped: # Can be decoded with bitwise flag with & 4 (4 means unmapped)
-        outfile.write("%s\t%s\t%s\t%s\t%s\t%s\t%s\n" % (read.reference_name, read.pos+51, info[4], nucl, info[5], 
-            info[6], info[7]))
+    # Getting AS and XS:
+    AS = read.get_tag("AS")
+    try:
+        XS = read.get_tag("XS")  # Some variants don't have secondary alignments, which throws an error
+    except KeyError:
+        XS = -int(flanklength) # Set an arbitrary low value for the "artificial" secondary alignment 
+
+    # Calculating correct position and filtering based on alignement around the variant position
+    start = read.pos
+    readVarPos = start + int(flanklength) + 1
```

Comment:
PEP8: camelCase is not used in Python for naming variables; it should be `ready_var_pos` instead (here and throughout the Python modules)

https://www.python.org/dev/peps/pep-0008/#function-and-variable-names